backends/memcache.go

```diff
@@ -33,30 +52,28 @@ func NewMemcacheBackend(cfg config.Memcache) *Memcache {
 	} else {
 		mc = memcache.New(cfg.Hosts...)
 	}
-	c.client = mc
-	return c
+
+	return &MemcacheBackend{
+		memcache: &Memcache{mc},
+	}
 }
 
-func (mc *Memcache) Get(ctx context.Context, key string) (string, error) {
-	res, err := mc.client.Get(key)
+func (mc *MemcacheBackend) Get(ctx context.Context, key string) (string, error) {
+	res, err := mc.memcache.Get(key)
 
 	if err != nil {
+		if err == memcache.ErrCacheMiss {
+			err = utils.KeyNotFoundError{}
+		}
 		return "", err
 	}
```

Comment:
Nitpick: Could be written as:
```
if err == memcache.ErrCacheMiss {
  err = utils.KeyNotFoundError{}
} else if err != nil {
  return "", err
}
```
.. not sure if this looks cleaner to you.

Reply:
I'm a bit split on this one. I agree that the `if/else` looks cleaner but I kind of like the fact that we test for `nil` first. You decide, I'm ok either way.

Reply:
Scott's above code is broken. If it is an `ErrCacheMiss` error, then the error won't be returned, as the else will skip `return "", err`.

Reply:
Ohh I see, it needs a `return` statement, right?
```
   if err == memcache.ErrCacheMiss {
-      err = utils.KeyNotFoundError{}
+      return utils.KeyNotFoundError{}
   } else if err != nil {
       return "", err
   }
```